Bilibili: Some extractor improvements

asgembed opened the pull request.

asgembed committed:
bilibili: use API for `article_json` fetching

Using page url may get CAPTCHA, so switch it to API.
This method is more likely to cause `-352` error when I testing..?

asgembed committed:
bilibili: commentary fetching when opus API failed

Originally from pull #6138.

Add opus commentary support for `post_json` so that the commentary can still be fetched when the opus API fails.

asgembed committed:
bilibili: support quote-post with image posts

https://danbooru.donmai.us/forum_topics/33428

"Not sure what to call it" again.

hdk5 commented:
In case you are interested in improving `article_commentary_desc` to support all possible article formatting, you can test on this article 

https://www.bilibili.com/opus/1127456471773085717

asgembed committed:
bilibili: `buvid3` for opus API

This can reduce the chance of getting a -352 error.

asgembed committed:
bilibili: refactor article commentary desc

Support more article formatting.

Also fix line breaks in `t.bilibili.com/:id` posts.

asgembed committed:
bilibili: ignore some `[Image]` in commentary

asgembed committed:
bilibili: `CGI.escapeHTML` and list line breaks fx

And new short link domain for Bilibili?

hdk5 commented on app/logical/source/extractor/bilibili.rb:
```diff
@@ -22,86 +21,190 @@ def image_urls
+        urls = []
+
+        urls += post_json.dig("modules", "module_dynamic", "major", "opus", "pics").to_a.pluck("url")
+        urls += post_json.dig("modules", "module_dynamic", "desc", "rich_text_nodes").to_a.select do |node|
```
Not really sure if applicable here, but the order of extracted images should be the same as they appear in the post
